companion/src/storage/categorized.cpp

```diff
@@ -254,9 +254,34 @@ bool CategorizedStorageFormat::loadYaml(RadioData & radioData)
 
   QByteArray modelslistBuffer;
   if (loadFile(modelslistBuffer, "MODELS/models.yml")) {
-    if (!loadModelsListFromYaml(radioData.categories, modelFiles, modelslistBuffer)) {
-      setError(tr("Can't load MODELS/models.yml"));
-      return false;
+    try {
+      if (!loadModelsListFromYaml(radioData.categories, modelFiles, modelslistBuffer)) {
+        setError(tr("Can't load MODELS/models.yml"));
```

Comment:
If there is a missing YML (i.e. B&W or simply missing), yes this seems to work. But if there is a corrupt radio.yml or models.yml, instead you only get

https://github.com/EdgeTX/edgetx/blob/5aa866acc26a7fd37fd6524ec10e609878182ac1/companion/src/radiointerface.cpp#L213-L216

Rather than this line (259) for the models (or 263) or in the case of a corrupt/invalid model file

https://github.com/EdgeTX/edgetx/blob/5aa866acc26a7fd37fd6524ec10e609878182ac1/companion/src/storage/categorized.cpp#L298-L306

Reply:
btw, the same radiointerface.cpp:213 also seems to fire when you answer no to the radio vs profile mismatch question... but I don't know whether that is a good or bad thing... I seem to think a more specific message should be given in response to that answer and that message not be shown then. Just mentioning whilst thinking of it...

Reply:
I tried with a corrupted `.etx` where I purposely corrupted `RADIO/radio.yml` and got this:
<img width="422" alt="Screenshot 2022-01-08 at 16 25 53" src="https://user-images.githubusercontent.com/1050031/148649866-340e417b-46c9-4d24-92c4-2bec73c17429.png">

The segfault issue has been fixed.

Reply:
Thank you, yes, it's gone ;) Reading from `.etx` isn't an issue, reading from the radio is ;) i.e. a corrupted on RADIO/radio.yml on actual radio will give... 
![image](https://user-images.githubusercontent.com/5500713/148665224-8c90d7e0-9cb6-43c5-9660-a4464ef0c6f5.png)

Reply:
Same for encountering a corrupt model file: 
![image](https://user-images.githubusercontent.com/5500713/148665276-56dc1d93-d534-4c54-8239-925672d1e7ef.png)

Reply:
No yaml-cpp error ;)

Reply:
Ok, `yaml-cpp` errors should be displayed properly now. `readSettingsSDCard` was simply ignoring them and displaying it's own error message instead (unconditionally).